src/Microsoft.Performance.SDK/Processing/FileDataSource.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     Represents a file backed data source for process.
+    /// </summary>
+    public class FileDataSource
+        : DataSource
+    {
+        /// <summary>
+        ///     Initializes a new <see cref="FileDataSource"/> with the specified file path.
+        /// </summary>
+        /// <param name="path">
+        ///     The path to the file.
+        /// </param>
+        public FileDataSource(string path)
+            : base(new Uri(Path.GetFullPath(path)))
+        {
+            Guard.NotNullOrWhiteSpace(path, nameof(path));
```

Comment:
>Guard.NotNullOrWhiteSpace(path, nameof(path)); [](start = 12, length = 46)

We haven't been consistent about documenting exceptions. So I'm taking this opportunity to ask... should we try and maintain <exception> doc comments? (I hate myself for asking this.)